### PR TITLE
Frr api url param

### PR DESCRIPTION
**Pull request by rfrancel**
@gjn here is a [test  link](https://mf-chsdi3.dev.bgdi.ch/frr_tilematrixset/shorten/6bcb421dfb) with the chsdi branch as url parameter

linked to https://github.com/geoadmin/mf-geoadmin3/issues/3290

### CODE REVIEW

File: src/index.mako.html [hunk 1]
```diff
@@ -726,22 +726,34 @@
         var cacheAdd = '${version}' != '' ? '/' + '${version}' : '';
         var pathname = location.pathname.replace(/(index|mobile|embed)\.html$/g, '');
 
+        function getParam(name){
+          return decodeURIComponent((new RegExp('[?|&]' + name + '=' + '([^&;]+?)(&|#|;|$)').exec(location.search)||[,""])[1].replace(/\+/g, '%20'))||null;
```
Comment (gjn) [inline src/index.mako.html]: Curios: where is the regexp coming from?

File: src/index.mako.html [hunk 2]
```diff
@@ -726,22 +726,34 @@
...
+        }
+
+        var my_api_url = getParam('api_url');
+
+        if (my_api_url){
+          api_url = '//mf-chsdi3.dev.bgdi.ch/' + my_api_url;
```
Comment (gjn) [inline src/index.mako.html]: We should think about integration, or other environments as well. So I would rather have here the domain of the ${api_url} variables per default. Then optionally, we can add a 2nd paramter where we can specify the environment. So we would have something like:

```
var my_api_env = getParam('api_env');
api_url = domain_from_url(${api_rul);
if (my_api_env) {
  api_url = '//mf-csdhi3.' + my_api_env + 'bgdi.ch/' + my_api_url;
}

**Comment by gjn** [comment]
Well done. Exactly what I had in mind. I've added a comment for potential improvement.

Also, there's one thing we need to address:
LayersConfig and Topic services are _not_ served with the api_url (this is by design, because otherwise, offline won't work).

But what we could do is that if we detect an api_url parameter in the URL, we serve the LayersConfig and the Topic services from the api_url as well.

See https://github.com/geoadmin/mf-geoadmin3/blob/master/src/index.mako.html#L782 for layersConfig url configuration and https://github.com/geoadmin/mf-geoadmin3/blob/master/src/index.mako.html#L786 for topics url functionality.

File: src/index.mako.html (hunk 1 above)
Comment (rfrancel) [inline src/index.mako.html]: the regexp helps us find the parameter value. What is wrong with it? Is there another method you prefer I'd use?

**Comment by rfrancel** [comment]
@gjn  I changed the PR according to your comments:
Here is a [test link](https://mf-geoadmin3.dev.bgdi.ch/frr_apiUrl_param/?api_env=int&topic=ech&lang=fr&X=189918.22&Y=659918.22&zoom=1&bgLayer=ch.swisstopo.swissimage) with the api_env parameter
and here is [one](https://mf-geoadmin3.dev.bgdi.ch/frr_apiUrl_param/?api_url=frr_shop_tooltips&topic=ech&lang=fr&X=190000.00&Y=660000.00&zoom=1&bgLayer=ch.swisstopo.swissimage&layers=ch.swisstopo.zeitreihen,ch.bfs.gebaeude_wohnungs_register,ch.bafu.wrz-wildruhezonen_portal,ch.swisstopo.swisstlm3d-wanderwege&layers_visibility=false,false,false,false&layers_timestamp=18641231,,,) woth a branch parameter.
I am not happy about the 'slice the url if it ends with /' but the base api_url does not return the same in my dev environment or in my deployed branch. 
//mf-chsdi3.dev.bgdi.ch in my env
//mf-chsdi3.dev.bgdi.ch/ in my deployed branch
Since we will probably be using these parameters in both cases, and that I could not find the reason for this difference, I did it like that..

File: src/index.mako.html (hunk 1 above)
Comment (gjn) [inline src/index.mako.html]: Nothing wrong with it. I just wanted to know where you got it from - or if you came up with it yourself.

**Comment by gjn** [comment]
@rfrancel Very nice. Please squash the commits and merge.

File: src/index.mako.html (hunk 1 above)
Comment (rfrancel) [inline src/index.mako.html]: Ah no I found it here: http://stackoverflow.com/questions/11582512/how-to-get-url-parameters-with-javascript/11582513#11582513

**Comment by rfrancel** [comment]
Thank you very much I'll merge when the tests are done

**Pull request merged**